components/coding/SkillRatingsComponent.tsx

```diff
@@ -0,0 +1,122 @@
+import { useQuery, useMutation, useLazyQuery } from "@apollo/client";
+import { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  FetchUserSkillsRatings,
+  FETCH_USER_SKILLS_RATINGS,
+  UserSkillsRatings,
+} from "../../graphql/fetchUserSkillsRatings";
+import { UPSERT_USER_SKILL_RATINGS } from "../../graphql/upsertUserSkillRatings";
+import { useAuth } from "../../lib/authContext";
+import {
+  skillRatingsSelector,
+  setSkillRatings,
+  SkillRatingsRow,
+} from "../../redux/skillRatingsSlice";
+import SkillSection from "../skillRatings/SkillSection";
+import { Button } from "../ui/Button";
+import { FetchAllSkills, FETCH_ALL_SKILLS } from "./FetchAllSkills";
+
+const initializeSkillRating = (skillIds: string[], userId: string) => {
+  const newSkillRatings = skillIds.map((skillId) => {
+    return {
+      skillId: skillId,
+      userId: userId,
+      studentRating: 0,
+    };
+  });
+
+  return newSkillRatings;
+};
+
+export default function SkillRatingsComponent(props) {
+  const dispatch = useDispatch();
+  const { user } = useAuth();
+  const { skillRatings } = useSelector(skillRatingsSelector);
+
+  const {} = useQuery<FetchAllSkills>(FETCH_ALL_SKILLS, {
+    onCompleted: (data) => {
+      const skillIds = data.intro_course_skills.map((it) => it.id);
+
+      setSkillIds(skillIds);
+    },
+  });
+  const [skillIds, setSkillIds] = useState<string[]>([]);
+  const [currentSkills, setCurrentSkills] = useState<string[]>([]);
+
+  const {} = useQuery<FetchUserSkillsRatings>(FETCH_USER_SKILLS_RATINGS, {
+    variables: {
+      userId: user.uid,
+    },
+    onCompleted: (data) => {
+      const currentSkillsFiltered = data.intro_course_skills_user.filter(
+        (it) => it.intro_course_skill.id
+      );
+      console.log("CSF", currentSkillsFiltered);
+      const currentSkillsMapped = data.intro_course_skills_user.map(
```

Comment:
naming conventions are generally confusing, can we do something like: allSkills, currentlyRatedSkills, missingSkills

Reply:
changed to as suggested.